### DISCUSSION TITLE
How to get source image without comment like highlight, text etc.?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#3138] [General]
How to get source image without comment like highlight, text etc.?
Below method will include comments on exported image, how to improve?

                    pix = page.get_pixmap(matrix=fitz.Matrix(300/72.0, 300/72.0))

**Comment by JorjMcKie** [comment]
This an image **_created_** from the whole page. Nothing gets "exported" here.
You cannot influence what is shown on it - except for making an image of only some part of the page (`clip` parameter).

**Comment by nissansz** [comment]
How to remove all comments in a pdf?

**Reply by JorjMcKie** [reply]
I don't understand: Which comments?
The comments inside the page's appearance stream - those prefixed with "%"?
I need an example.

**Reply by JorjMcKie** [reply]
Did you mean the `Document` method `scrub()`?

**Comment by nissansz** [comment]
Examples
[澳门科技大学 交通資訊.pdf](https://github.com/pymupdf/PyMuPDF/files/14194051/default.pdf)

![image](https://github.com/pymupdf/PyMuPDF/assets/41010669/6894a00c-5756-4eab-a64a-4133d3263d66)

**Reply by JorjMcKie** [reply]
Ah - these seem to text marker annotations. They are no "comments"!

Please look at the [get_pixmap](https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_pixmap) documentation: you can exclude annotations from being shown in the page pixmap.
Annotations can also be deleted completely - if this is what you meant.

**Comment by nissansz** [comment]
Thank you.
annots=False is what I need
get_pixmap(*, matrix=fitz.Identity, dpi=None, colorspace=fitz.csRGB, clip=None, alpha=False, annots=True)
